src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/RewindDataIndexRepository.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using DBreeze;
+using DBreeze.DataTypes;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.ProvenBlockHeaders
+{
+    /// <summary>
+    /// Persistent implementation of the <see cref="RewindData"/> DBreeze repository.
+    /// </summary>
+    public class RewindDataIndexRepository : IRewindDataIndexRepository
+    {
+        /// <summary>
+        /// Instance logger.
+        /// </summary>
+        private readonly ILogger logger;
+
+        /// <summary>
+        /// Access to DBreeze database.
+        /// </summary>
+        private readonly DBreezeEngine dbreeze;
+
+        /// <summary>
+        /// DBreeze table names.
+        /// </summary>
+        private const string RewindDataIndexTable = "RewindDataIndex";
+
+        /// <summary>
+        /// Initializes a new instance of the object.
+        /// </summary>
+        /// <param name="folder"><see cref="ProvenBlockHeaderRepository"/> folder path to the DBreeze database files.</param>
+        /// <param name="loggerFactory">Factory to create a logger for this type.</param>
+        public RewindDataIndexRepository(string folder, ILoggerFactory loggerFactory)
+        {
+            Guard.NotNull(folder, nameof(folder));
+
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+
+            Directory.CreateDirectory(folder);
+
+            this.dbreeze = new DBreezeEngine(folder);
```

Comment:
Should we not reuse the same transaction from DbreezeCoinview?

Reply:
How do we do this? All the disk access is hidden away in DbreezeCoinview and we don't have access to it.

Reply:
Wouldn't we then tightly couple rewind data index store to a specific implementation of coin view - DBreezeCoinView. Maybe we should consider moving RewindDataIndexStore as a dependency to `DBreezeCoinView` so that DBreezeCoinView uses `RewindDataIndexStore` to create and manage rewind data and its index?